packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -79,7 +79,10 @@ function buildHierarchy(
 
         // Add in either the blob or tree
         if (entry.type === "tree") {
-            const newTree = { id: entry.id, blobs: {}, commits: {}, trees: {} };
+            const newTree: api.ISnapshotTree = { id: entry.id, blobs: {}, commits: {}, trees: {} };
+            if (entry.unreferenced) {
```

Comment:
I'd assign unconditionally as part of initial assignment line - it's fine to have undefined property and it's faster that way :)